### PR TITLE
Publish operator binaries from release script

### PR DESCRIPTION
**Pull request by lgarciaaco**
With this PR, when doing a release by running `syndesis release` (not a snapshot release), a [release in github](https://github.com/syndesisio/syndesis/releases) is created and the operator binaries are included.

---
How to use:
```
 syndesis release --release-version tag_name 
```
Replace tag_name with with the real tag, 1.9 ... for instance

---
What produces:
The result is similar to [this release](https://github.com/syndesisio/syndesis/releases/tag/1.9.99999). To the description, `git log` with commits added since the last release will be added, it should look like:
```
f36f417 - Merge branch 'master' into refactor/json-utils
e13475a - fix: warnings in Maven POM (#6994)
56e2884 - refactor: remove unused imports (#6992)
5a91bc7 - refactor: remove needless else (#6991)

```

The release will be created when `syndesis release` is ran without `--snapshot-release`

### CODE REVIEW

File: tools/bin/commands/release [hunk 1]
```diff
@@ -109,10 +106,16 @@ release::run() {
...
-    # Create the operator image
+    # Create the operator image binaries
     update_image_versions "$topdir" "$release_version" "$moving_tag"
     "$topdir/install/operator/build.sh" --operator-build docker --image-build docker --image-name "syndesis/syndesis-operator" --image-tag "$release_version"
 
+    # Prepare binaries for release
+    prepare_binaries "${topdir}/install/operator/dist" "${topdir}/install/operator/releases"
+
+    # Release the binaries
+    publish_artifacts "${topdir}/install/operator/releases" "$release_version"
```
Comment (zregvart) [inline tools/bin/commands/release]: I'd move towards the end, there's no guarantee that the tags, docker pushes and Maven central staging repository release will happen. No point in announcing a release without those happening before.

File: tools/bin/commands/release [hunk 2]
```diff
@@ -235,6 +232,71 @@ update_image_versions() {
...
+
+publish_artifacts() {
+    local release_dir=$1
+    local tag=$2
+
+    get_github_access_token
+    get_github_username
+
+    set +e
+    local upload_url=$(curl -q --fail -X POST -u $GITHUB_USERNAME:${GITHUB_ACCESS_TOKEN} -H "Accept: application/vnd.github.v3+json" -H "Content-Type: application/json" -d "{\"tag_name\": \"${tag}\"}" https://api.github.com/repos/syndesisio/syndesis/releases | jq -r .upload_url | cut -d{ -f1)
```
Comment (zregvart) [inline tools/bin/commands/release]: We could add `prerelease: true` to the payload here if this is a snapshot release.

**Comment by zregvart** [comment]
Would it make sense to specify `name` on the release payload, perhaps for snapshot builds from master `latest` would be a good name, others could derive from the tag.

**Comment by lgarciaaco** [comment]
Fixes https://github.com/syndesisio/syndesis/issues/6976

File: tools/bin/commands/release [hunk 3]
```diff
@@ -235,6 +234,105 @@ update_image_versions() {
...
+    local tag=$2
+
+    local release_name=$(get_gh_release_name ${tag})
+    local desc=$(get_gh_release_desc)
+
+    get_github_access_token
+    get_github_username
+
+    set +e
+    local upload_url=$(curl -q --fail -X POST -u $GITHUB_USERNAME:${GITHUB_ACCESS_TOKEN} -H "Accept: application/vnd.github.v3+json" -H "Content-Type: application/json" -d "{\"tag_name\": \"${tag}\", \"name\": \"${release_name}\", \"body\": \"${desc}\"}" https://api.github.com/repos/syndesisio/syndesis/releases | jq -r .upload_url | cut -d{ -f1)
```
Comment (zregvart) [inline tools/bin/commands/release]:
```suggestion
    local data="{
	  \"tag_name\": \"${tag}\",
	  \"name\": \"${release_name}\",
	  \"body\": \"${desc}\"}
	"
	local upload_url=$(curl -q --fail \
      -X POST \
      -u $GITHUB_USERNAME:$GITHUB_ACCESS_TOKEN \
      -H "Accept: application/vnd.github.v3+json" \
      -H "Content-Type: application/json" \
      -d "$data" \
      https://api.github.com/repos/syndesisio/syndesis/releases | jq -r '.upload_url | sub("{.*"; "")'
```

The payload `-d` bit could be a nice variable and parameters to curl could be on individual lines, to help with readability, `jq` can handle search-replace of strings.

File: tools/bin/commands/release [hunk 4]
```diff
@@ -235,6 +234,105 @@ update_image_versions() {
...
+        echo "ERROR: Cannot create release on remote github repository. Check if a release with the same tag already exists."
+        return
+    fi
+
+    for file in $release_dir/*; do
+        curl -q --fail -X POST -u $GITHUB_USERNAME:$GITHUB_ACCESS_TOKEN \
+          -H "Accept: application/vnd.github.v3+json" \
+          -H "Content-Type: application/tar+gzip" \
+          --data-binary "@$file" \
+          $upload_url?name=${file##*/} >$ERROR_FILE 2>&1
```
Comment (zregvart) [inline tools/bin/commands/release]: What's `$ERROR_FILE`?

File: tools/bin/commands/release [hunk 5]
```diff
@@ -235,6 +234,105 @@ update_image_versions() {
...
+        cp ${from}/${dist}/syndesis-operator ${to}/syndesis-operator-${dist}
+    done
+}
+
+get_gh_release_name() {
+    local tag=$1
+    local current_branch=$(git branch  | grep \* | awk '{print $2}')
+    local name=${tag}
+
+    if [[ "${current_branch}" == "master" ]]; then
```
[the remark this answers is not in the document]
Comment (lgarciaaco) [reply tools/bin/commands/release]: oh this doesnt make any sense ... the whole if should go

[the remark this answers is not in the document]
Comment (lgarciaaco) [reply tools/bin/commands/release]: Ok this is what I am thinking ... we will do releases from master as well as other branches ... how do we differentiate them? I thought that releases from master should be called: $tag-master .. . while other branches only the tag, and this is what I am doing.

ie. 
```
1.9-master
OR
1.8
```

**Pull request closed without merging**